src/index.ts

```diff
@@ -285,6 +285,12 @@ export function apply(ctx: Context, config: Config) {
       session.send(globalTasks.size
         ? session.text('.pending', [globalTasks.size])
         : session.text('.waiting'))
+      
+      if (config.globalConcurrency) {
+        while (globalTasks.size >= config.globalConcurrency) {
+          await sleep(100)
+        }
+      }
```

Comment:
Not a great implementation, I think it would be better that you have a `pending` list caching those pending request and then pick from it when you have requests resolved.